app/controllers/katello/api/v2/content_view_versions_controller.rb

```diff
@@ -33,6 +33,7 @@ def index_relation
       versions = versions.in_environment(@environment) if @environment
       versions = versions.with_puppet_module(@puppet_module) if @puppet_module
       versions = versions.component_of(params[:composite_version_id]) if params[:composite_version_id]
+      versions = versions.with_names(JSON.parse(params[:names])) if params[:names]
```

Comment:
Calling `JSON.parse` here seems odd. Either you're expecting a JSON encoded string within your POST request (probably should use an array instead) or in your JSON request (probably shouldn't send JSON inside JSON).